hs/rsh/stdlib.rsh

```diff
@@ -35,8 +35,10 @@ export const fromMaybe = (v, onNull, onSome) => {
   case None: return onNull();
   case Some: return onSome(v); } };
 
-export const isSome = m => fromMaybe(m, () => false, (_) => true);
-export const isNone = m => fromMaybe(m, () => true, (_) => false);
+// TODO - this definitely emits optimized teal code, but it no longer checks that the data is actually a proper Maybe type...
```

Comment:
Maybe a silly idea, but what about this for typechecking?
```
const isSome = m => {
  assert(m.match({ Some: true, None: true });
 return _dataTag(m) == 1;
}

Reply:
Thanks!  Good idea, although superseded by Jay's suggestion.